gcp/src/ARCFileReader.cxx

```diff
@@ -324,27 +343,38 @@ void ARCFileReader::ParseArrayMap(uint8_t *buf, size_t size)
 				memcpy(block_name, buf + off, namelen);
 				block_name[namelen] = '\0';
 				off += namelen;
-				
+
 				/* Block parameters */
-				memcpy(ltmp, buf + off, sizeof(ltmp));
-				off += sizeof(ltmp);
+				if (experiment == Experiments::BK) {
+					// BK has one fewer dim registers
+					memcpy(ltmp, buf + off, sizeof(ltmp) - sizeof(ltmp[0]));
+					off += sizeof(ltmp) - sizeof(ltmp[0]);
+				} else {
+					memcpy(ltmp, buf + off, sizeof(ltmp));
+					off += sizeof(ltmp);
+				}
 				block_info.flags = ntohl(ltmp[0]);
 				block_info.mode = ntohl(ltmp[1]);
 				block_info.base = ntohl(ltmp[2]);
 				block_info.addr = ntohl(ltmp[3]);
 				block_info.dim[0] = ntohl(ltmp[4]);
 				block_info.dim[1] = ntohl(ltmp[5]);
-				block_info.dim[2] = ntohl(ltmp[6]);
+				if ((experiment == Experiments::SPT) ||
+					(experiment == Experiments::PB)) {
+					block_info.dim[2] = ntohl(ltmp[6]);
+				} else {
+					block_info.dim[2] = 0;
+				}
```

Comment:
Should the sense of this if block be inverted (BK being the odd man out as at line 348)?